modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -996,6 +996,45 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
                              CV_OUT std::vector<int>& indices,
                              const float eta = 1.f, const int top_k = 0);
 
+     class CV_EXPORTS_W Model : public Net
+     {
+     public:
+         Model(const std::string& model, const std::string& config = "");
+         Model(const Net& network);
+
+         Model& setInputSize(const Size& size);
+         Model& setInputSize(int width, int height);
+         Model& setInputMean(const Scalar& mean);
+         Model& setInputScale(float scale);
+         Model& setInputCrop(bool crop);
+         Model& setInputSwapRB(bool swapRB);
+         void predict(InputArray frame, OutputArrayOfArrays outs);
+
+     protected:
+         struct Impl;
+         Ptr<Impl> impl_;
+     };
+
+     class CV_EXPORTS_W ClassificationModel : public Model
+     {
+     public:
+         ClassificationModel(const std::string& model, const std::string& config = "");
+         ClassificationModel(const Net& network);
+
+         std::vector<std::pair<int, float>> classify(InputArray frame);
```

Comment:
How we can interpret multiple outputs from classification model?

Reply:
Impl::predict return std::vector<Mat>& outs.Do we need to check that vector contains 1 element?

Reply:
Sure. We can have multiple outputs from detection networks because there are could be more than one object and in example YOLOv3 produces outputs for different objects sizes. However there is one and only prediction for classification problem.